docs: [TRG] Initial Proposal for a documentation TRG "Migration Guide" (1.09) (#1187)
eclipse-tractusx/eclipse-tractusx.github.io: docs/release/trg-1/trg-1-09.md

---
title: TRG 1.09 - Migration Guide
sidebar_position: 1
---

:::caution

> Proposed release date: "mandatory after": 19th of May 2023

Reviewer: ```suggestion
Proposed release date: "mandatory after": tbd
```

:::

| Status     | Created      | Post-History                           |
|------------|--------------|----------------------------------------|

> | Draft      | 12-March-2025 | Initial contribution                  |

Reviewer 2: I suggest to change the date to May, 16 2025 and the status to active: we'd merge it then during the committer meeting of  May, 16 2025

Author: Done.

## Why

Operators and users of a component need a way to understand the impact of a new release on their work. This could be a required database migration, a change in an API call or system behavior.

Those changes aren't obvious just based on a changelog ([TRG 1.03](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-3)) that usually only list all PRs within a release.

## Description

> A migration guide **must** contain all relevant information for operators of a component to migrate an existing component from the previous to the new version.

Reviewer 3: ```suggestion
A migration guide **must** contain all relevant information for operators of a component to migrate an existing component from the previous to the new version. It must be placed in a directory named `/docs/architecture'.
```

I would add a directory where it's expected. We've got the same for all other TRGs + we could extend the TRG conformance workflow that it at least checks, that it's given.

It **must** also contain all relevant changes for users, such as changed API calls, component behaviour, added/deprecated customer facing features etc.

The migration guide shall _explain_ relevant changes to operators and users.

> ## Best Practices

Reviewer 4: I really hate the fact, that TRGs are currently not very consistent in their structure.

20 TRGs use `Example`
5 TRGs Use  `Best Practice`
1 TRG uses `Best Practices and Example`

I suggest switching to `Example` and also including an example and scrapping `Best Practice`

Author: @ClosedSourcerer Examples and Best Practices are two different things for me. I chose not to provide an example, because this always depends on the application... We could however link to "good examples" that already do that, such as the portal or the EDC. I'd still like to keep the best practices... would you suggest to add them to the "description"?

> - A migration guide should be understandable by both operators and users of a component.

Reviewer 4: `IT Operations Management` is well established ITIL lingo and should be easily understood by the audience of this TRG.

> - A migration guide should be understandable by both operators and users of a component.

Author: @ClosedSourcerer @lgblaumeiser so you would propose something like "A migration guide should support IT operations management and enable usage of a new release"?

> - A migration guide should be understandable by both operators and users of a component.

Reviewer: It is not that easy, I have to admit
```suggestion
- A migration guide should support IT operations management and guide the efforts for updating dependent tools and processes.
```

> - There should be a dedicated migration guide per release: E.g. From 1.1.x -> 1.2.x, 1.2.x -> 2.0.x etc.

Reviewer 5: We could narrow the need for a migration guide to major releases only, which is the case where breaking changes can occur.

Author: @rafaelmag110 we aren't really following semantic versioning in all products - EDC and DTR contain "breaking breaking changes" although we just increase the minor counter. So a proposal could be "There must be a dedicated migration guide for each major release. For minor releases a guide must only be provided if it affects the usage or operation of a service.
